thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/data/entity/QuestionsSet.java

```diff
@@ -0,0 +1,43 @@
+package pl.edu.agh.quizzesthesis.data.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.hibernate.annotations.GenericGenerator;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import java.util.Objects;
+
+@Entity
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class QuestionsSet {
+
+    @Id
+    @GeneratedValue(generator = "questions-set-key-generator")
+    @GenericGenerator(name = "questions-set-key-generator",
+            strategy = "pl.edu.agh.quizzesthesis.QuestionsSetKeyGenerator")
+    private String questionsSetKey;
+
+    private String questionsSetName;
+    private Integer teacherId;
```

Comment:
Is there a reason you use user's nick and id separately as fields instead of
```
@ManyToOne
private User user
```
?

Reply:
The reason is that we get this values from `@Current`, so I just putted them there. If I understand correctly, I would need to query for a `User` to get a `user`, but I still only use `teacherId` for now (`nick` **can** be used if we decide to). So to sum up, the advantages are the reduced amount of queries and no need for `UserRepository`, the disadvantage is data redundancy. Please review if my opinion is reasonable or do you still want me to change this, in that case I will just change it.

Reply:
You are right, I changed it so that now we have relation between `QuestionsSet` and `Users`.